### ISSUE TITLE
About hoverfly configuration persistence

### ISSUE THREAD

**Issue by navy1997** [SpectoLabs/hoverfly#1163]
Does hoverfly support configuration persistence? If we need to make hoverfly into multiple copies, if there is no data persistence, it seems that we need to create the configuration of each copy

**Comment by navy1997** [comment]
I tried to start boltdb. But after restarting, I found that the configuration I wrote was empty again.
![1734682543261](https://github.com/user-attachments/assets/74cff646-7d27-47eb-b672-531af58d7ae0)

**Comment by tommysitu** [comment]
I don't think hoverfly supports config persistence. You may want to try out https://hoverfly.io/hoverfly-cloud

**Comment by navy1997** [comment]
If I want hoverfly to restart without losing my previously configured request config, is it not supported?

**Comment by navy1997** [comment]
> I don't think hoverfly supports config persistence. You may want to try out https://hoverfly.io/hoverfly-cloud

If I want hoverfly to restart without losing my previously configured request config, is it not supported?

**Comment by tommysitu** [comment]
oh you meant the simulation data? You can export it into a json file before you shut down hoverfly, and import it in again after a restart.

**Comment by navy1997** [comment]
> simulation data

Yes, it is simulated data. Doesn't the -db parameter mean writing the configuration to disk? I specify the db path when I restart, and then read the previous configuration. It seems that it is not supported at present.

**Comment by tommysitu** [comment]
the db flag was used for storing admin user and tokens, unfortunately they have been deprecated:

```
  -db string
    	Storage to use - 'boltdb' or 'memory' which will not write anything to disk (DEPRECATED) (default "memory")
  -db-path string
    	A path to a BoltDB file with persisted user and token data for authentication (DEPRECATED)
```

**Comment by Najafov007** [comment]
Hello! Is there anything new on this topic? We need to somehow save this data in disk where we have already PVC. How can we persist this data

**Comment by tommysitu** [comment]
you can save the simulation data by exporting them to a JSON file

**Comment by Najafov007** [comment]
> you can save the simulation data by exporting them to a JSON file

But there is no actually hoverfly commands to export it - can you share please how is it done please?

**Comment by tommysitu** [comment]
@Najafov007 you can use `hoverctl`, check this out: https://docs.hoverfly.io/en/stable/pages/tutorials/basic/exportingsimulations/exportingsimulations.html
